fe/fe-core/src/main/java/org/apache/doris/plugin/dialect/HttpDialectConverterPlugin.java

```diff
@@ -100,10 +123,43 @@ public ImmutableSet<Dialect> acceptDialects() {
     @Override
     public @Nullable String convertSql(String originSql, SessionVariable sessionVariable) {
         String targetURL = GlobalVariable.sqlConverterServiceUrl;
-        if (Strings.isNullOrEmpty(targetURL)) {
+
+        if (Strings.isNullOrEmpty(targetURL) && dialectConverterSerInfo.isEmpty()) {
             return null;
         }
-        return HttpDialectUtils.convertSql(targetURL, originSql, sessionVariable.getSqlDialect());
+
+        if (!Strings.isNullOrEmpty(targetURL)) {
+            String resultStr = HttpDialectUtils.convertSql(targetURL, originSql, sessionVariable.getSqlDialect());
+            if (Strings.isNullOrEmpty(resultStr)) {
+                return originSql;
+            } else {
+                return resultStr;
+            }
+        } else {
+            if (!sessionVariable.isEnableMultiDialectConvertService()) {
+                return originSql;
+            }
+
+            int retryCount = 0;
+            while (retryCount < dialectConverterSerInfo.size()) {
+                String chooseServer = dialectConverterSerInfo.get(chooseServerIdx++ % dialectConverterSerInfo.size());
+                retryCount++;
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("convertsql server is : {}", chooseServer);
+                }
+
+                targetURL = "http://" + chooseServer + "/api/v1/convert";
+                String resultStr = HttpDialectUtils.convertSql(targetURL, originSql,
+                        sessionVariable.getSqlDialect());
+
+                if (Strings.isNullOrEmpty(resultStr)) {
+                    continue;
+                } else {
+                    return resultStr;
+                }
+            }
+            return originSql;
```

Comment:
Should we log here if the conversion fails? 
The log should contain how many times tried and which server has been tried.